Source/iOS/Utilities.swift

```diff
@@ -0,0 +1,13 @@
+import UIKit
+
+public struct Utilities {
+
+  public static var isPushNotificationEnabled: Bool {
+    guard let settings = UIApplication.shared.currentUserNotificationSettings else {
+      return false
+    }
+
+    return UIApplication.shared.isRegisteredForRemoteNotifications
```

Comment:
This is not enough on iOS 10. It only checks if the app is _registered_ for notifications, not if they are actually enabled. To check that, you need to call `UNUserNotificationCenter.getNotificationSettings()`, and check the setting's `authorizationStatus`.